packages/app/src/visualizer/utils.ts

```diff
@@ -36,9 +41,24 @@ export function resolvePath(
 ): { entity: ProvidedEntity; supportedVis: VisDef[] } | undefined {
   const entity = entitiesStore.get(path);
 
-  const supportedVis = findSupportedVis(entity, attrValuesStore);
-  if (supportedVis.length > 0) {
-    return { entity, supportedVis };
+  if (isDataset(entity)) {
+    const supportedVis = getSupportedCoreVis(entity, attrValuesStore);
+    return supportedVis.length > 0 ? { entity, supportedVis } : undefined;
+  }
+
+  if (!isGroup(entity)) {
+    return undefined;
+  }
+
+  if (isNxNoteGroup(entity, attrValuesStore)) {
+    return { entity, supportedVis: [NX_NOTE_VIS] };
+  }
```

Comment:
I now consider `NXnote` and `NXdata` groups as distinct entity "kinds" at the top level of the algorithm. This will help me refactor `getSupportedNxDataVis` much more easily.